cmake/compiler/xcc-clang/compiler_flags.cmake

```diff
@@ -6,6 +6,18 @@ include(${ZEPHYR_BASE}/cmake/compiler/clang/compiler_flags.cmake)
 set_compiler_property(PROPERTY nostdinc)
 set_compiler_property(PROPERTY nostdinc_include)
 
+# For C++ code, set nostdinc_include compiler flag to contain
+# path to the compiler defined header files.
```

Comment:
> I think you could avoid low-level implementation details and negations and focus on the intention and higher-level effect like this:
> 
> Of course the next question is: why was this include removed in the first place and why is this needed only by `xcc-clang`? I found the answer neither in the comments nor in the commit message.

@marc-hb So I updated the comment and the commit message here. Let me know if additional changes are required. Thanks!!

Reply:
>  The solution implemented here is the latter one for ease of clarity for the reader.

I bet this is outside the control of this PR but your lengthy explanation sounds like this #include situation is not going to get anywhere close to "clarity" any time soon. So a comment longer than "# For C++ code, re-add the standard includes directory" would not hurt. It is probably impossible for that comment to describe the entire situation "clearly" without growing too big. Even an incomplete comment with just clues and pointers would be much better; at the very least it would not fool the reader into thinking the situation is "clear" and would at least help them "come prepared".

Reply:
@Marc Updated the comments as per your suggestion.

Reply:
Very good, thank you! The greater picture is still too big for me to understand right now but now you're giving great starting points for anyone who either can or must spend the time to understand it.